share/sysbuild/Kconfig

```diff
@@ -41,4 +41,9 @@ config WARN_DEPRECATED
 	  Print a warning when the Kconfig tree is parsed if any deprecated
 	  features are enabled.
 
+config COMPILER_WARNINGS_AS_ERRORS
+	bool "Treat warnings as errors"
+	help
+	  Turn on "warning as error" toolchain flags for all images if set.
+
```

Comment:
Proposal, not required.

Should we already now place this in a build (tools) / compiler specific Kconfig file ?
I suspect this won't be the last setting of this kind.

Reply:
My proposal would be to consider #66253 and using `-Dall_CONFIG_COMPILER_WARNINGS_AS_ERRORS=y`.
In general, I don't know if it makes sense for sysbuild to control toolchain settings.

Reply:
> My proposal would be to consider https://github.com/zephyrproject-rtos/zephyr/pull/66253 and using -Dall_CONFIG_COMPILER_WARNINGS_AS_ERRORS=y.

yes, that would be a good solution for this specific use-case.
But we still have some outstanding issues related to #66253, my biggest concern related to Kconfig settings which takes a relative path. Relative in context of an `all_CONFIG_SETTING` would be problematic.

So until we have decided how to solve such cases, then I believe this PR is an acceptable solution to the issue at hand.

> In general, I don't know if it makes sense for sysbuild to control / be aware of toolchain settings.

I agree there.
Especially because different images could be building with different compilers / architecture specific flags, and therefore would be target (image) specific settings.

But there are few settings, like warnings as errors, which are generic enough and thus make sense to share.

Reply:
Moved to `build` folder